fix_checkpoint_history.py

```
@@ -0,0 +1,80 @@
+    if target_csv not in history:
+        history[target_csv] = []
+    
+    history[target_csv].append({
+        "checkpoint_file": str(checkpoint_file),
+        "location_index": estimated_locations,
+        "timestamp": datetime.now().isoformat(),
+        "total_locations": 1607,
+        "measurements_count": line_count
+    })
+    
+    with open(history_file, 'w') as f:
```

Pylint (3.3.7)

[warning] Using open without explicitly specifying an encoding

(W1514)